### PR TITLE
Add runtimeconfig.json to BuiltProjectOutputGroupOutput

### LINKED ISSUE #472: runtimeconfig.json file is not added to BuiltProjectOutputGroupOutput in inner build
For any app targeting netcoreapp1.0 , a runtimeconfig.json file is generated in the bin folder, but pack fails to include it in the nupkg because this is not included in the list of files given by BuiltProjectOutputGroup target.

Talked to @rainersigwald about this, the bug lies in 
https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets#L119-L128 

This issue is being tracked on the NuGet side by : https://github.com/NuGet/Home/issues/3878

CLI : https://github.com/dotnet/cli/issues/2432

### PR DESCRIPTION
**Pull request by dsplaisted**
Adds runtimeconfig.json to BuiltProjectOutputGroupOutput item. This will cause it to be included in the NuGet package if an application project is packed.

Fixes #472

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets [hunk 1]
```diff
@@ -131,6 +131,16 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!-- Add runtimeconfig.json file to BuiltProjectOutputGroupOutput, so that it will get included in the NuGet package by the Pack target -->
+  <Target Name="AddRuntimeConfigFileToBuiltProjectOutputGroupOutput"
+          Condition=" '$(GenerateRuntimeConfigurationFiles)' == 'true'"
+          BeforeTargets="BuiltProjectOutputGroup"
+          DependsOnTargets="GenerateBuildRuntimeConfigurationFiles">
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Should this depend on actually doing the generation?

**Review by srivatsn — APPROVED** [review]

**Comment by srivatsn** [comment]
@dsplaisted can you add the template?

**Comment by dsplaisted** [comment]
@MattGertz for approval

Escrow template:

#### Scenario

Create a NuGet package from a .NET Core app

#### Bug

#472

#### Workarounds

Adding the following to your project file would probably work:

```xml
 <!-- Add runtimeconfig.json file to BuiltProjectOutputGroupOutput, so that it will get included in the NuGet package by the Pack target -->
  <Target Name="AddRuntimeConfigFileToBuiltProjectOutputGroupOutput"
          Condition=" '$(GenerateRuntimeConfigurationFiles)' == 'true'"
          BeforeTargets="BuiltProjectOutputGroup">
    <ItemGroup>
      <BuiltProjectOutputGroupOutput Include="$(ProjectRuntimeConfigFilePath)" FinalOutputPath="$(ProjectRuntimeConfigFilePath)" />
    </ItemGroup>
  </Target>
```

#### Risk

Low

#### Performance Impact

Low

#### Regression Analysis

Not a regression

**Comment by natidea** [comment]
Approved to merge via [378318](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/378318)

**Pull request merged**